### PR TITLE
Allow retry parsing source code in case of timeouts

### PR DESCRIPTION
**Pull request by neakor**
Also refactored the entry point to use better code reuse, and added unit tests for the entry points.

### CODE REVIEW

File: Generator/Sources/NeedleFramework/Entry/PluginizedGenerator.swift [hunk 1]
```diff
@@ -0,0 +1,31 @@
+//
```
Comment (rudro) [inline Generator/Sources/NeedleFramework/Entry/PluginizedGenerator.swift]: Should this one file live in a `Pluginized` subdirectory ? Seems silly for one file, but it's the convention we've used everywhere else.

File: Generator/Sources/NeedleFramework/Utilities/Constants.swift [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+/// The name of the NeedleFoundation module.
+let needleModuleName = "NeedleFoundation"
+
+/// The default value for waiting timeout.
```
Comment (rudro) [inline Generator/Sources/NeedleFramework/Utilities/Constants.swift]: nit: add (in seconds) ?

Comment (rudro) [reply Generator/Sources/NeedleFramework/Utilities/Constants.swift]: oh, wait.  where is this used?

File: Generator/Sources/NeedleFramework/Entry/PluginizedGenerator.swift (hunk 1 above)
Comment (neakor) [reply Generator/Sources/NeedleFramework/Entry/PluginizedGenerator.swift]: Yeah I think it'll be too many folders. Let's keep all the entry stuff in one folder for now, unless more shows up.

File: Generator/Sources/NeedleFramework/Utilities/Constants.swift (hunk 2 above)
Comment (neakor) [reply Generator/Sources/NeedleFramework/Utilities/Constants.swift]: Sure. I'll add the unit. This is used in a bunch of places where we wait for the executor to finish.

**Pull request merged**